diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -1379,6 +1415,12 @@ int DifferentialFunctionComparator::cmpFieldAccess(
     if (!(GEPL && GEPR))
         RETURN_WITH_LOG(1);
 
+    // If we previously ignored allocation of memory that any of the two GEPs
+    // point to, fail (cmpValues would compare their pointer operands as equal).
+    if (hasIgnoredAllocaAsPtrOperand(GEPL)
+        || hasIgnoredAllocaAsPtrOperand(GEPR))
+        RETURN_WITH_LOG(1);
```

Comment:
What is the purpose of this? It can lead to false positives so we double-check why it is necessary.

Reply:
I didn't figure it out. It might have been relevant before I added some other check elsewhere, but everything seems to work fine without this now.